components/automate-ui/src/app/entities/userperms/userperms.effects.ts

```diff
@@ -17,27 +28,41 @@ import {
   GetUserParamPermsFailure
 } from './userperms.actions';
 import { UserPermsRequests } from './userperms.requests';
+import { getlastFetchTime } from './userperms.selectors';
 
 @Injectable()
 export class UserPermEffects {
   constructor(
     private actions$: Actions,
-    private requests: UserPermsRequests
+    private requests: UserPermsRequests,
+    private store: Store<NgrxStateAtom>
   ) { }
 
-  // include debounce here because each request of this type is a true duplicate
+  // the target limit
+  private freshLimitInMilliseconds = 30000;
+
+  // to account for a flurry of IntrospectSome calls almost concurrent
+  // with our IntrospectAll call--this will happen frequently!--just
+  // extend the fresh limit by a nudge for the IntrospectSome calls
+  private freshLimitPlusANudge = 33000;
+
   @Effect()
   fetchAllPerms$ = this.actions$.pipe(
     ofType(UserPermsTypes.GET_ALL),
-    debounceTime(20),
-    mergeMap(() => this.requests.fetchAll().pipe(
-      map((resp: UserPermsResponsePayload) => new GetAllUserPermsSuccess(resp.endpoints)),
-      catchError((error: HttpErrorResponse) => of(new GetAllUserPermsFailure(error))))));
+    throttleTime(this.freshLimitInMilliseconds),
```

Comment:
(3) The throttling here serves as the cache expiry control for IntrospectAll itself.